metadata-ingestion/src/datahub/ingestion/masking/bootstrap.py

```diff
@@ -0,0 +1,461 @@
+                "urllib3.connectionpool",
+                "urllib3.util.retry",
+                "requests",
+            ]:
+                try:
+                    logging.getLogger(logger_name).setLevel(logging.INFO)
+                except Exception:
+                    pass
+
+            _bootstrap_completed = True
+            _bootstrap_error = None
+            secret_count = registry.get_count()
+            logger.info(
+                f"Secret masking initialized successfully for {context.value} "
+                f"with {secret_count} secrets registered"
+            )
+
+        except Exception as e:
+            _bootstrap_error = e
+            logger.error(f"Failed to initialize secret masking: {e}", exc_info=True)
+            # Don't raise - graceful degradation
+
+
+def _load_secrets_for_context(
+    context: ExecutionContext,
+    registry: SecretRegistry,
+    explicit_sources: Optional[List[str]] = None,
+) -> None:
+    """
+    Load secrets from appropriate sources based on context.
+
+    Args:
+        context: Execution context
+        registry: SecretRegistry instance
+        explicit_sources: Optional list of sources to load
+    """
+    # Determine sources
+    sources = explicit_sources if explicit_sources else _get_default_sources(context)
+
+    logger.debug(f"Loading secrets from sources: {sources}")
+
+    for source in sources:
+        try:
+            if source == "environment":
+                _load_environment_secrets(registry)
+            elif source == "datahub_secrets_store":
+                # TODO: Implement when backend API available
+                logger.debug("DataHub secrets store not yet implemented")
+            else:
+                logger.warning(f"Unknown secret source: {source}")
+        except Exception as e:
+            logger.warning(f"Failed to load secrets from {source}: {e}")
+
+
+def _get_default_sources(context: ExecutionContext) -> List[str]:
+    """
+    Determine secret sources based on execution context.
+
+    All contexts use environment variables as the base, since DataHub
+    substitutes ${VAR} with os.environ['VAR'] in recipe files.
+
+    Args:
+        context: Execution context
+
+    Returns:
+        List of source names to load
+    """
+    # Base: All contexts mask environment variables
+    sources = ["environment"]
+
+    # Additional: UI and remote may use DataHub secrets store in future
+    if context in [ExecutionContext.UI_BACKEND, ExecutionContext.REMOTE_EXECUTOR]:
+        # TODO: Implement datahub_secrets_store loading when backend API is available
+        # sources.append("datahub_secrets_store")
+        pass
+
+    return sources
+
+
+# Environment variables that should NOT be masked (common system variables)
+#
+# Note: Never add variables containing secrets, passwords, keys, or tokens.
+#       Examples to NEVER add: AWS_ACCESS_KEY_ID, DATABASE_PASSWORD, API_KEY,
+#       SECRET_KEY. If unsure, don't add it.
+_SYSTEM_ENV_VARS = {
```

Comment:
We may rename with a more descriptive name such as: 
```
DATAHUB_MASKING_ENV_VARS_ALLOWED
DATAHUB_MASKING_ENV_VARS_SKIPPED
...
```
or something like that

Additionally, we may have a mechanism to incorporate values without requiring a release:

```
DATAHUB_MASKING_ENV_VARS_ALLOWED_PATTERN
```

so we can provide a regex pattern to skip additional env vars